### PR TITLE
Map votes now draw from a hat

### PR DESCRIPTION
**Pull request by Rukofamicom**
## About The Pull Request

When the server holds a map vote, it no longer automatically choses the map with the most votes, but instead pools all of the votes together and picks one person's vote randomly. Map with the most votes is the most likely to be picked, but the map picked will not always be Metastation when a vote is held. This allows maps to be variable and ensures they all see some play, but also allows players to most frequently play on popular maps that more people like. 

## Why It's Good For The Game
Enabling votes resolves the issue of being locked into the same map over and over... except that it sometimes doesn't because then it would nearly always be the same map - the one with the most support every time a vote is called. This allows player votes to matter in deciding next map without guaranteeing only the single most popular map sees play. Stations which are less popular will still show up (so long as at least one person votes for it) but they won't be there as often.  

## Testing Photographs and Procedure

Tested by @Mat05usz to ensure it works properly. 

## Changelog
:cl: Mat05usz, Rukofamicom
tweak: Map votes are now weighted by number of votes rather than picking the map with the most votes
/:cl:

### CODE REVIEW

**Comment by Mat05usz** [comment]
This is a better solution than full RNG map rotation. Also as discussed with Ruko, after certain time has passed (maybe 45 minutes, maybe more/less), the map that's currently played will be excluded from vote, but that has yet to be implemented. 

Why is this a better solution: 

1. If round ends too quickly (nukies, maybe wizard) you can still vote for the same map
2. If round lasts long enough, current map will not be votable to allow players choose another one

I believe that's the most fair and least frustrating solution to the current problem of playing the same map 3/4/more times in a row, because RNG decided it feels cute today.

**Comment by Mat05usz** [comment]
Would be nice if people who downvote said why do they think current RNG based map rotation, that frequently doesn't change maps because yes, is better than letting players decide which map is most probable to be picked.

**Comment by Mat05usz** [comment]
As a sidenote, I did test the pickweight and it does work, sometimes picking options that didn't get the most votes.

**Comment by Isy232** [comment]
Epic, now instead of when 55% of people want meta and being stuck with getting 100% meta.

**Comment by PowerfulBacon** [comment]
Should this be a config option?

**Comment by Rukofamicom** [comment]
How many of our downstreams have map voting enabled?
Or do they all have the same issue we did where it would just always be the most popular map?

I'd normally say being an option is always better, but I'm genuinely not sure in this case.

**Comment by Mat05usz** [comment]
Can we at least get it testmerged or should we suffer by having the same map picked 10 times in a row?

**Comment by Rukofamicom** [comment]
give it another week or so, things will pick up. 

(Also they already merged like 20 things today, too much at once is asking for trouble)

File: code/modules/client/preferences.dm [hunk 1]
```diff
@@ -1708,7 +1708,7 @@ GLOBAL_LIST_EMPTY(preferences_datums)
 						default += " ([config.defaultmap.map_name])"
 					for (var/M in config.maplist)
 						var/datum/map_config/VM = config.maplist[M]
-						if(!VM.votable)
+						if(!VM.votable || (SSmapping.config.map_name == VM.map_name && REALTIMEOFDAY - SSticker.round_start_time >= 24000)) //current map will be excluded from the vote if round has lasted more than 40 minutes.
```
Comment (ike709) [inline code/modules/client/preferences.dm]: Map rotation excludes the current map entirely and IMO that should also be the case here.

Comment (Rukofamicom) [reply code/modules/client/preferences.dm]: Addressed to match review

**Review by ike709 — APPROVED** [review]

**Pull request merged**